### PR TITLE
[FEAT] Goal Service

### PR DESCRIPTION
**Pull request by sueelly**
## 📝 Purpose of PR

Goal Service를 구현한다.

## Contents

Command, Query는 결국 api 요청에 대해서 수행된다. 그 어디에도 포함되지 않는 기능들의 경우 기본 Service로 생성하고, 이것을 사용하도록 구현했다. 또한 주 데이터베이스는 결국 command가 수행되는 것이고, 이는 일반 service에서도 접근 가능하므로 일반 repository로 정정했다.
기본적인 기능을 분리함에 따라 commandService에서도 핵심 로직에만 집중할 수 있다.

- [x] unit test
- [x] Implement Goal Service and apply it
- [x] GoalCommandRepository -> GoalRepository

## Checklist

-   [x] Does commit messages follow the convention?
-   [x] Are variable names brief but descriptive?
-   [x] Is the code clean and easy to understand?
-   [x] If a new feature has been added, or a bug fixed, has a test been added to confirm good behavior?
-   [x] Does the test successfully test edge and corner cases?

## 💬 Review Requirements (Optional)

Is there anything you can praise about this PR? Start with praise.
Make a review, leaving kind comments and suggesting changes where needed (to resolve the above).

### CODE REVIEW

File: src/main/java/ac/dnd/dodal/application/goal/service/GoalService.java
Comment (5hseok) [inline src/main/java/ac/dnd/dodal/application/goal/service/GoalService.java:1]: 이렇게 Service를 따로 정의해두고 CommandService에서 의존성을 주입받는 방식이 깔끔해보이네요!

**Review by 5hseok — APPROVED** [review]

File: src/main/java/ac/dnd/dodal/application/goal/repository/GoalRepository.java [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import ac.dnd.dodal.domain.goal.model.Goal;
 
 @Repository
-public interface GoalCommandRepository extends JpaRepository<Goal, Long> {
+public interface GoalRepository extends JpaRepository<Goal, Long> {
```
Comment (5hseok) [inline src/main/java/ac/dnd/dodal/application/goal/repository/GoalRepository.java:11]: save랑 find로 조회하는 것은 Command와 Query의 차이가 아닌가요? 
GoalCommandRepository가 그냥 GoalRepository로 된 이유를 좀 더 자세히 듣고 싶어요! 
일반 서비스에서 접근하는 레포지토리여서 그런걸까요?

Comment (sueelly) [reply src/main/java/ac/dnd/dodal/application/goal/repository/GoalRepository.java:11]: command, query의 경우 결국 요청사항에 대한 부분이고, 이외에도 비즈니스 로직상에서 필요한 기능들도 존재합니다. 이 때문에 기본 service가 탄생한거고요.
cqrs를 사용한다고 하여도 결국 '주' db는 command용 db이고, 기본 service에서 사용하는 것 또한 마찬가지입니다.
crud 모두 해당 db에 수행됩니다. 조회 또한 많이 일어납니다. 단, 반환용이 아닌거죠.
따라서 command 용 db라고 지정하는 것보단 주 db로 사용하고 query만 따로 GoalQueryRepository로 분리하는 게 낫다고 생각했습니다!
